components/PralineForm/index.js

```diff
@@ -0,0 +1,214 @@
+import React, { useState } from "react";
+import { uid } from "uid";
+import useSWR from "swr";
+import { CldUploadButton } from "next-cloudinary";
+import Image from "next/image";
+import IngredientList from "./IngredientList";
+import InputField from "./InputField";
+
+export default function ProductForm() {
+  const [ingredients, setIngredients] = useState([]);
+  const [allergyTraces, setAllergyTraces] = useState([]);
+  const [currentPraline, setCurrentPraline] = useState(null);
+  const [nameField, setNameField] = useState("");
+  const [versionField, setVersionField] = useState("");
+  const [weightField, setWeightField] = useState("");
+  const [imageId, setImageId] = useState(null);
+
+  function handleAddIngredient(newIngredient) {
+    setIngredients([...ingredients, { ...newIngredient, id: uid() }]);
+  }
+
+  function handleDeleteIngredient(id) {
+    setIngredients(ingredients.filter((zutat) => zutat.id !== id));
+  }
+
+  function handleAddAllergyTraces(allergyTraceWithAmount) {
+    setAllergyTraces([
+      ...allergyTraces,
+      { ...allergyTraceWithAmount, id: uid() },
+    ]);
+  }
+
+  function handleDeleteAllergyTrace(id) {
+    setAllergyTraces(
+      allergyTraces.filter((allergyTrace) => allergyTrace.id !== id)
+    );
+  }
+
+  function cancel() {
+    setIngredients([]);
+    setAllergyTraces([]);
+    setCurrentPraline(null);
+    setNameField("");
+    setVersionField("");
+    setWeightField("");
+  }
+
+  const { data, isLoading, mutate } = useSWR("/api/pralinen");
+
+  if (isLoading) {
+    return <h1>Loading...</h1>;
+  }
+
+  if (!data) {
+    return null;
+  }
+
+  async function handleSubmit(event) {
+    event.preventDefault();
+
+    if (ingredients.length === 0) {
+      return;
+    }
+
+    const formData = new FormData(event.target);
+    const pralineData = Object.fromEntries(formData);
+
+    const newPraline = {
+      name: pralineData.name,
+      version: pralineData.version,
+      weight: pralineData.weight,
+      imageId: imageId,
+      ingredients: ingredients,
+      allergyTraces: allergyTraces,
+    };
+
+    const response = await fetch("/api/pralinen", {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(newPraline),
+    });
+
+    if (response.ok) {
+      mutate();
+    }
```

Comment:
This logic would need to go up as well. You can create a function on the page level and pass the function down in order to call it here.

The reason why you'd do that is the `mutate` function among other reasons. After you lift up the data fetching and the list, `mutate` is gone from this component.